Manuals/Altibase_trunk/eng/JDBC User's Manual.md

```diff
@@ -2247,6 +2273,93 @@ This method operates when the ResultSet object is of the following types:
 
 If this method is called for a TYPE_FORWARD_ONLY type, an exception is raised; for a TYPE_SCROLL_INSENSITIVE type, nothing happens.
 
+### Statement Caching
+
+*Statement Caching* is a feature designed to enhance performance by caching and reusing SQL statements when they are frequently executed. This functionality can be enabled without requiring modifications to the existing program. PreparedStatement and CallableStatement objects are the targets of caching, while Statement objects are not cached. Statement Caching is managed and configured at the Connection level.
+
+#### Basic Operation
+
+When executing the close() method of the Statement interface, a statement is cached. Subsequently, upon executing the prepareStatement() or prepareCall() methods, the system searches for an identical statement within the cached statements. If a matching statement is found in the cache, it returns the corresponding Statement object; otherwise, it creates a new Statement object and returns it.
+
+##### Criteria for Identical Statements
+
+Conditions under which a statement is considered identical to a cached statement include:
+
+- *SQL Consistency*: The SQL statement must be identical.
+- *Statement Type Matching*: The statement type (e.g., PreparedStatement or CallableStatement) must match.
+- *ResultSet Attributes*: Attributes of ResultSet objects generated by the statement (such as Scrollable, Concurrency, and Holdability) must be consistent.
+
+##### Handling of Cached Statements
+
+In situations where Statement Caching is enabled, calling the close() method of the Statement interface does not result in the physical closure of the statement. Instead, it is cached for potential reuse.
```

Comment:
When statement caching is enabled, the close() method of the statement interface caches the statement instead of closing it. That is, the statement does not physically close.

Reply:
네, 이게 훨씬 더 좋네요~